source-academy/frontend
# Resolve loading indicator issues in grading table

Resolves #3048

Good to do before tackling #2814 so that loading can be more explicity indicated.

This PR resolves the loading indicator bug in the grading table, whereby a request for a new submissions page does not properly get reflected as "loading" until backend data arrives. This problem likely arised due to the use of a deprecated API from AGGrid.

To accompany this change, additional measures have been added to prevent users from spamming new requests while awaiting for data - this disabling is now made possible by the proper loading state.

### Description



### Type of change


### How to test



- In the backend, set a high number of students in `priv/repo/seeds.exs`, example 20000 students and 100 questions. It should be enough such that backend fetching takes at least 1 second.
- Go to the grading table and attempt to load the next page. Notice that the loading logo stays until new data is received.
- Attempt to "spam" the next page button. Notice that you are now unable to do so due to the page change locking while data is being loaded.
- Attempt to continuously hit the refresh button. Notice that it is also disabled while data is being loaded.
- Note that this lock is not applied on the dropdown list as there is no HTML API to disable it easily - but this shouldn't be an issue since it cannot be repeatedly clicked in a short time.

### Checklist
